axelrod/strategies/lookerup.py

```diff
@@ -0,0 +1,124 @@
+    all as strings. The values are the moves to play on this round.  For
+    example, if
+
+        - the opponent started by playing C  
+        - my last move was a C the opponents
+        - last move was a D
+
+    the corresponding key would be
+
+        ('C', 'C', 'D')
+
+    and the value would contain the move to play on this turn.
+
+    Some well-known strategies can be expressed as special cases; for example
+    Cooperator is given by the dict:
+
+        {('', '', '') : 'C'}
+
+    where m and n are both zero. Tit For Tat is given by:
+
+    { 
+        ('', 'C', 'D') : 'D',         
+        ('', 'D', 'D') : 'D',         
+        ('', 'C', 'C') : 'C',
+        ('', 'D', 'C') : 'C',     
+    }
+
+    where m=1 and n=0.
+    
+    """
+    
+
+    name = 'LookerUp'
+    classifier = {
+        'memory_depth': float('inf'), 
+        'stochastic': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self, lookup_table):
+        """
+        """
+        Player.__init__(self)
+        self.lookup_table = lookup_table
+
+        # rather than pass number of previous rounds to consider in as a separate variable, figure it out
+        # the number of rounds is the length of the second element of any given key in the dict 
+        self.plays = len(self.lookup_table.keys()[0][1]) 
+
+        # the number of opponent starting moves is the lgnth of the first element of any given key in the dict
+        self.opponent_start_plays = len(self.lookup_table.keys()[0][0])
+
+        if self.opponent_start_plays == 0:
+            self.classifier['memory_depth'] = self.plays
+
+        self.init_args = ()
+
+    def strategy(self, opponent):
+        """
+        If there isn't enough history to lookup a move from the table, cooperate
+        """
+        if len(self.history) < max([self.plays,self.opponent_start_plays]) :
+            return 'C'
+            
+        else:
+            # count back m moves to get my own recent history
+            history_start = -1 * self.plays
+            my_history = ''.join(self.history[history_start:])
+
+            # do the same for the opponent
+            opponent_history = ''.join(opponent.history[history_start:])
+
+            # get the opponents first n moves
+            opponent_start = ''.join(opponent.history[0:self.opponent_start_plays])
+
+            # put these three strings together in a tuple
+            key = (opponent_start, my_history, opponent_history)
+
+            # look up the move associated with that tuple in the lookup table
+            move = self.lookup_table[key]
+
+            return move
+
+
+class EvolvedLookerUp(LookerUp):
+    """
+    A LookerUp strategy that uses a lookup table generated using an evolutionary algorithm. 
+    //TODO: update this docstring with a link to a blog post once I've written about it :-)
```

Comment:
I think it would be really great to get the blog post linked before merging. Would that be possible?

If not perhaps a reference or url to a brief explanation of evo algs... (I'm being picky here, happy to chat about it).